### DISCUSSION TITLE
NextJS 14 window is not defined

### DISCUSSION THREAD

**Discussion by alvarolozano** [tolgee/tolgee-platform#2670] [Q&A]
I followed the whole guide to setup the tolgee for NextJS app router (https://docs.tolgee.io/js-sdk/integrations/react/next/app-router) and when the detectLanguageFromHeaders is invoked, it gives me the following error
![image](https://github.com/user-attachments/assets/8758286a-7147-4b30-b6fe-5cdd66773f45)
Also, if I omit that part and set the language directly to de default it is working. 

By researching in the code, I found that the DetectLanguage method called from the detectLanguageFromHeaders reads the window object, and as it is a server side code, it fails
![image](https://github.com/user-attachments/assets/772e0743-2799-47dd-b460-4c3b475490a1)

**Comment by stepan662** [comment]
Thanks for reporting, I'll investigate.

**Reply by stepan662** [reply]
It's a bug, I've created a fix here https://github.com/tolgee/tolgee-js/pull/3401
